node/src/actors/session/handlers.rs

```diff
@@ -244,14 +258,38 @@ impl StreamHandler<BytesMut, Error> for Session {
                                                     );
                                                 }
                                             }
-
                                             // Stop block sending if an error occurs
                                             break;
                                         }
                                     }
                                 }
 
-                                actix::fut::ok(())
+                                actix::fut::ok(send_superblock_votes)
+                            })
+                            .and_then(|send_superblock_votes, session, _ctx| {
+                                let chain_manager_addr = ChainManager::from_registry();
```

Comment:
This section needs a comment explaining what are we doing and why do we need to do it. For example:

```suggestion
                            .and_then(|send_superblock_votes, session, _ctx| {
                                // If this is the last batch, send to the peer all the superblock votes that are currently stored in
                                // ChainManager. This allows faster synchronization in some cases, because if a node has not
                                // received enough votes, it will revert to the last consolidated superblock and start the
                                // synchronization again.
                                // Note that it is not strictly needed as part of the protocol.
                                let chain_manager_addr = ChainManager::from_registry();
```

Reply:
Sounds right! :+1: